bigtable/tests/unit/test_row_data.py

```diff
@@ -529,6 +529,7 @@ def test_cancel(self):
         self.assertEqual(response_iterator.cancel_calls, 0)
         yield_rows_data.cancel()
         self.assertEqual(response_iterator.cancel_calls, 1)
+        self.assertRaises(StopIteration, iter(yield_rows_data).next())
```

Comment:
We should be testing what we expect callers to want to see.  E.g., once `cancel` is called, iteration stops, yielding an empty list:

```python
        yield_rows_data.cancel()
        self.assertEqual(response_iterator.cancel_calls, 1)
        self.assertEqual(list(yield_rows_data), [])
```

Reply:
Agreed, although one could imagine a scenario when the list can be empty for reasons other than canceling the iteration. Done.